From 6279d04292000261edb77be40678bd3cf5489481 Mon Sep 17 00:00:00 2001
From: "Darryl L. Pierce" <mcpierce@apache.org>
Date: Fri, 7 Feb 2014 13:44:03 +0000
Subject: [PATCH] QPID-5499: Fix Ruby/Perl bindings when built with
 -Werror=format-security

Changed the swig descriptors so that they use a constant format string.

git-svn-id: https://svn.apache.org/repos/asf/qpid/trunk@1565651 13f79535-47bb-0310-9956-ffa450edef68
---
 qpid/cpp/bindings/qmf2/ruby/ruby.i         |  2 +-
 qpid/cpp/bindings/qpid/perl/perl.i         |  2 +-
 qpid/cpp/bindings/qpid/ruby/ruby.i         | 42 +++++++++++++++---------------
 qpid/cpp/include/qpid/swig_perl_typemaps.i |  2 +-
 qpid/cpp/include/qpid/swig_ruby_typemaps.i |  2 +-
 5 files changed, 25 insertions(+), 25 deletions(-)

--- a/bindings/qmf2/ruby/ruby.i
+++ b/bindings/qmf2/ruby/ruby.i
@@ -28,7 +28,7 @@
     }
     catch (qpid::types::Exception& mex) {
         static VALUE qmferror = rb_define_class("QmfError", rb_eStandardError);
-        rb_raise(qmferror, mex.what());
+        rb_raise(qmferror, "%s", mex.what());
     }
 }
 
--- a/bindings/qpid/perl/perl.i
+++ b/bindings/qpid/perl/perl.i
@@ -27,7 +27,7 @@
         $action
     }
     catch (qpid::messaging::MessagingException& mex) {
-        Perl_croak(aTHX_ mex.what());
+      Perl_croak(aTHX_ "%s", mex.what());
     }
 }
 
--- a/bindings/qpid/ruby/ruby.i
+++ b/bindings/qpid/ruby/ruby.i
@@ -28,7 +28,7 @@
     }
     catch (qpid::messaging::MessagingException& mex) {
         static VALUE merror = rb_define_class("MessagingError", rb_eStandardError);
-        rb_raise(merror, mex.what());
+        rb_raise(merror, "%s", mex.what());
     }
 }
 
--- a/bindings/swig_perl_typemaps.i
+++ b/bindings/swig_perl_typemaps.i
@@ -110,7 +110,7 @@
             }
             }
         } catch (qpid::types::Exception& ex) {
-            Perl_croak(aTHX_ ex.what());
+          Perl_croak(aTHX_ "%s", ex.what());
         }
 
         return result;
--- a/bindings/swig_ruby_typemaps.i
+++ b/bindings/swig_ruby_typemaps.i
@@ -106,7 +106,7 @@
             }
         } catch (qpid::types::Exception& ex) {
             static VALUE error = rb_define_class("Error", rb_eStandardError);
-            rb_raise(error, ex.what());
+            rb_raise(error, "%s", ex.what());
         }
 
         return result;
